zerver/lib/dev_ldap_directory.py

```diff
@@ -34,15 +34,15 @@ def generate_dev_ldap_dir(mode: str, num_users: int=8) -> Dict[str, Dict[str, An
             email_username = email.split('@')[0]
             ldap_dir['uid=' + email_username + ',ou=users,dc=zulip,dc=com'] = {
                 'cn': [name[0], ],
-                'userPassword': email_username,
+                'userPassword': [email_username, ],
```

Comment:
Looks great, but let's add a failing login test for this with the invalid value :).